diff --git a/lcls1_producers/smd_producer.py b/lcls1_producers/smd_producer.py
index 4c2fe87..dcdf7d4 100755
--- a/lcls1_producers/smd_producer.py
+++ b/lcls1_producers/smd_producer.py
@@ -654,7 +654,7 @@ for evt_num, evt in enumerate(ds.events()):
 
     #the ARP will pass run & exp via the enviroment, if I see that info, the post updates
     if ( (evt_num<100 and evt_num%10==0) or (evt_num<1000 and evt_num%100==0) or (evt_num%1000==0)):
-        if os.environ.get('ARP_JOB_ID', None) is not None:
+        if os.environ.get('ARP_JOB_ID', None) is not None and os.getenv("JID_UPDATE_COUNTERS") is not None:
             if ds.size == 1:
                 requests.post(os.environ["JID_UPDATE_COUNTERS"], json=[{"key": "<b>Current Event</b>", "value": evt_num+1}])
             elif ds.rank == 0:
@@ -703,7 +703,7 @@ if ds.rank==0:
 
 #finishing up here....
 logger.debug('rank {0} on {1} is finished'.format(ds.rank, hostname))
-if os.environ.get('ARP_JOB_ID', None) is not None:
+if os.environ.get('ARP_JOB_ID', None) is not None and os.getenv("JID_UPDATE_COUNTERS") is not None:
     if ds.size > 1:
         if ds.rank == 0:
             requests.post(os.environ["JID_UPDATE_COUNTERS"], json=[{"key": "<b>Last Event</b>", "value": "~ %d cores * %d evts"%(ds.size,evt_num)},{"key": "<b>Duration</b>", "value": "%f min"%(prod_time)}])
diff --git a/lcls2_producers/smd_producer.py b/lcls2_producers/smd_producer.py
index 9f52b00..2b50c9b 100755
--- a/lcls2_producers/smd_producer.py
+++ b/lcls2_producers/smd_producer.py
@@ -1058,6 +1058,6 @@ logger.debug("rank {0} on {1} is finished".format(rank, hostname))
 #        cmd = ['timestamp_sort_h5', h5_f_name, h5_f_name]
 
 if ds.unique_user_rank():
-    if os.environ.get("ARP_JOB_ID", None) is not None:
+    if os.environ.get("ARP_JOB_ID", None) is not None and os.getenv("JID_UPDATE_COUNTERS") is not None:
         requests.post(
             os.environ["JID_UPDATE_COUNTERS"],
             json=[
diff --git a/smalldata_tools/ana_funcs/azav_pyfai.py b/smalldata_tools/ana_funcs/azav_pyfai.py
index fafbced..1facbd4 100644
--- a/smalldata_tools/ana_funcs/azav_pyfai.py
+++ b/smalldata_tools/ana_funcs/azav_pyfai.py
@@ -5,9 +5,10 @@ try:
 except ModuleNotFoundError:
     print("pyFAI not available on LCLS-II env")
 
-from smalldata_tools.common.detector_base import DetObjectFunc
-from smalldata_tools.utilities import image_from_dxy
+from LCLSGeom.manager import get_geometry
+from LCLSGeom.converter import PsanaToPyFAI
 
+from smalldata_tools.common.detector_base import DetObjectFunc
 
 class azav_pyfai(DetObjectFunc):
     """
@@ -49,39 +50,16 @@ class azav_pyfai(DetObjectFunc):
     def __init__(self, **kwargs):
         self._name = kwargs.get("name", "pyfai")
         super(azav_pyfai, self).__init__(**kwargs)
-
         self.mask = kwargs.pop("userMask", None)
         self.threshold = kwargs.pop("thres", None)
         self.return2d = kwargs.pop("return2d", "False")
-
-        # azimuthal integrator argument
         self.poni_file = kwargs.pop("poni_file", None)
-        if self.poni_file is not None:
-            self.ai = pyFAI.load(str(self.poni_file))
-        else:
-            self._ai_kwargs = kwargs.pop("ai_kwargs", None)
-            assert (
-                self._ai_kwargs is not None
-            ), "Need either a calibration file or a set of keywords arguments to instantiate the pyFAI integrator."
-            # self.ai = pyFAI.azimuthalIntegrator.AzimuthalIntegrator(**self._ai_kwargs) # why does it not work?
-            self.ai = pyFAI.AzimuthalIntegrator(**self._ai_kwargs)
-        print(f"Created azimuthal integrator for {self._name}")
-        # we should have code that extract setup parameters
-        #    in case people mess w/ the input
-        # we should also extra q/theta/... bins for plotting later.
-        azcfg = self.ai.get_config()
-        dcfg = azcfg.pop("detector_config")
-        ocfg = dcfg.pop("orientation")
-        for k, v in azcfg.items():
-            setattr(self, "azcfg_%s" % k, v)
-        for k, v in dcfg.items():
-            setattr(self, "azcfg_det_%s" % k, v)
-        setattr(self, "azcfg_detorientation", ocfg.value)
-        setattr(self, "azcfg_qvals", self.ai.get_qa())
-        # setattr(self,'azcfg_qvals',self.ai.get_qa())
+        self._ai_kwargs = kwargs.pop("ai_kwargs", None)
+        if self._ai_kwargs is None and self.poni_file is None:
+            raise ValueError("Either a poni file or ai_kwargs must be provided to initialize the azimuthal integrator.")
 
         # integration arguments
-        self.pol_factor = kwargs.pop("polarization_factor", 1)
+        self.pol_factor = kwargs.pop("polarization_factor", -1)  # Vertical Polarization by default post Run 18
         self.npts = kwargs.pop("npts", 256)
         self.npts_az = kwargs.pop("npts_az", 360)
         # self.radial_range = kwargs.pop('radial_range',None)
@@ -94,27 +72,6 @@ class azav_pyfai(DetObjectFunc):
 
     def setFromDet(self, det):
         super(azav_pyfai, self).setFromDet(det)
-        # geometry
-        self.ix = det.ix
-        self.iy = det.iy
-
-        if self.poni_file is None:  # use calib value if calib file is given
-            if len(det.pixelsize) == 2:
-                if det.pixelsize[0] > 1:
-                    self.ai.pixel1 = det.pixelsize[0] * 1e-6
-                    self.ai.pixel2 = det.pixelsize[1] * 1e-6
-                else:
-                    self.ai.pixel1 = det.pixelsize[0]
-                    self.ai.pixel2 = det.pixelsize[1]
-            else:
-                if det.pixelsize[0] > 1:
-                    self.ai.pixel1 = det.pixelsize[0] * 1e-6
-                    self.ai.pixel2 = det.pixelsize[0] * 1e-6
-                else:
-                    self.ai.pixel1 = det.pixelsize[0]
-                    self.ai.pixel2 = det.pixelsize[0]
-
-        # use cmask by default. Note: pyFAI uses 0 for valid pixel.
         if (
             self.mask is not None
             and det.cmask is not None
@@ -135,12 +92,56 @@ class azav_pyfai(DetObjectFunc):
                     except:
                         self.mask = None
         if self.mask is not None and self.mask.ndim == 3:
-            maskInv = ~(self.mask.astype(bool))
-            self.mask = image_from_dxy(self.mask, self.ix, self.iy)
-            maskInvImg = image_from_dxy(maskInv, self.ix, self.iy)
-            maskImg = ~(maskInvImg)
-            self.mask = maskImg
-
+            self.mask = np.reshape(self.mask, (self.mask.shape[0]*self.mask.shape[1], self.mask.shape[2]))
+
+        metrology = get_geometry(det.det.alias)
+        # instantiate PyFAI detector first
+        # this adds the detector name mentioned in the poni file to pyFAI detector registry
+        # reading the poni file will not throw an error if the detector name is not in the registry
+        detector = PsanaToPyFAI.convert(metrology, det.det.alias)
+        # if detector is 3D (panels, slow, fast), then this scripts creates a fake 2D detector by
+        # squeezing the panel and slow dimensions.
+        # the 3D pixel layout is passed using the metrology data !
+        if self._ai_kwargs is None:
+            ai = pyFAI.load(self.poni_file)
+            self._ai_kwargs = ai.getPyFAI()
+
+        if self._units == "q_A^-1":
+            if "wavelength" not in self._ai_kwargs:
+                raise ValueError("Wavelength must be provided when using q_A^-1 units. Check poni file or ai_kwargs inputs.")
+            self.ai = pyFAI.AzimuthalIntegrator(
+                detector=detector,
+                dist=self._ai_kwargs["dist"],
+                poni1=self._ai_kwargs["poni1"],
+                poni2=self._ai_kwargs["poni2"],
+                rot1=self._ai_kwargs["rot1"],
+                rot2=self._ai_kwargs["rot2"],
+                rot3=self._ai_kwargs["rot3"],
+                wavelength=self._ai_kwargs["wavelength"],
+            )
+        else:
+            self.ai = pyFAI.AzimuthalIntegrator(
+                detector=detector,
+                dist=self._ai_kwargs["dist"],
+                poni1=self._ai_kwargs["poni1"],
+                poni2=self._ai_kwargs["poni2"],
+                rot1=self._ai_kwargs["rot1"],
+                rot2=self._ai_kwargs["rot2"],
+                rot3=self._ai_kwargs["rot3"],
+            )
+        # we should have code that extract setup parameters
+        #    in case people mess w/ the input
+        # we should also extra q/theta/... bins for plotting later.
+        azcfg = self.ai.get_config()
+        dcfg = azcfg.pop("detector_config")
+        ocfg = dcfg.pop("orientation")
+        for k, v in azcfg.items():
+            setattr(self, "azcfg_%s" % k, v)
+        for k, v in dcfg.items():
+            setattr(self, "azcfg_det_%s" % k, v)
+        setattr(self, "azcfg_detorientation", ocfg.value)
+        setattr(self, "azcfg_qvals", self.ai.get_qa())
+        # setattr(self,'azcfg_qvals',self.ai.get_qa())
         print(f"Azimuthal integrator:\n{self.ai}")
         return
 
@@ -149,9 +150,11 @@ class azav_pyfai(DetObjectFunc):
 
     def process(self, data):
         if self.threshold is not None:
-            data[data < threshold] = 0
+            data[data < self.threshold] = 0
         if data.ndim == 3:
-            data = image_from_dxy(data, self.ix, self.iy)
+            # instead of assembling the image, now we just squeeze the panel and slow dimensions
+            # so that image and detector have same shape.
+            data = np.reshape(data, (data.shape[0]*data.shape[1], data.shape[2]))
         out = self._process(data)
         return out
 
@@ -178,5 +181,4 @@ class azav_pyfai(DetObjectFunc):
                 method="cython",
                 **self._azav_kwargs,
             )
-
             return {"azav": I, "q": q}
diff --git a/summaries/BeamlineSummaryPlots_mfx.py b/summaries/BeamlineSummaryPlots_mfx.py
index dbd21ac..12a1b1f 100755
--- a/summaries/BeamlineSummaryPlots_mfx.py
+++ b/summaries/BeamlineSummaryPlots_mfx.py
@@ -125,6 +125,11 @@ logger = logging.getLogger(__name__)
 
 # Define Args
 parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--smd_path",
+    help="path to smalldata hdf5 file (def <exp>/hdf5/smalldata/<exp>_Run<run:04d>.h5)",
+    default=None,
+)
 parser.add_argument(
     "--run", help="run", type=str, default=os.environ.get("RUN_NUM", "")
 )
@@ -169,10 +174,13 @@ if int(os.environ.get("RUN_NUM", "-1")) > 0:
 ### load data for the chosen run  ####
 ######################################
 
-smd_path = (
-    f"/sdf/data/lcls/ds/mfx/{args.experiment}/hdf5/smalldata/"
-    f"{args.experiment}_Run{run:04d}.h5"
-)
+if args.smd_path:
+    smd_path = args.smd_path
+else:
+    smd_path = (
+        f"/sdf/data/lcls/ds/mfx/{args.experiment}/hdf5/smalldata/"
+        f"{args.experiment}_Run{run:04d}.h5"
+    )
 h5 = h5py.File(smd_path)
 
 ## Defining initial selection (laser-on events)
@@ -297,7 +305,7 @@ try:
     ).options(color="r")
     ipmDownTimeMed = hv.Points(
         (eventTimeRMed, ipmDownMed),
-        kdims=[eventTimeDim, ipmUpDim],
+        kdims=[eventTimeDim, ipmDownDim],
         label=ipmDownDim.label,
     ).options(color="m")
 
@@ -321,10 +329,10 @@ else:
     lxtPlot = None
 
 gspec = pn.GridSpec(
-    sizing_mode="stretch_both", max_width=700, name="Data Quality - Run %d" % run
+    sizing_mode="stretch_both", max_width=1000, max_height=1000, name="Data Quality - Run %d" % run
 )
-gspec[0:2, 0:8] = pn.Column(ipmTimeLayout)
-gspec[2:5, 0:4] = pn.Column(ipmLayout)
+gspec[0:4, 0:4] = pn.Column(ipmTimeLayout)
+gspec[0:4, 4:8] = pn.Column(ipmLayout)
 
 detNames: list = [
     "jungfrau",
@@ -343,7 +351,7 @@ for detname in detNames:
         damageVar = h5[damageDim.name][evtFilter]
         damagePlot = rolling(
             hv.Curve(damageVar, vdims=[damageDim], label=f"{detname}").opts(
-                axiswise=True, color=hv.Palette("Spectral")
+                axiswise=True, color=hv.Palette("Spectral"), title=f"Damage {detname}"
             ),
             rolling_window=10,
         )
@@ -351,12 +359,20 @@ for detname in detNames:
         continue
     plots.append(damagePlot)
 
-multiDamagePlot = hv.Overlay(plots).opts(
-    xlabel="Event (rolling average of 10)",
-    ylabel="Present (Yes/No)",
-    title="Missing/Damaged Data",
+damagePanels = (
+    hv.Layout(plots)
+    .opts(
+        hv.opts.Curve(
+            xlabel="Event (rolling average of 10)",
+            ylabel="Present (Yes/No)",
+            axiswise=True,
+        )
+    )
+    .cols(len(plots))
+    if plots
+    else hv.Curve([]).opts(title="No Damage Data")
 )
-gspec[2:5, 4:8] = multiDamagePlot
+gspec[4:6, 0:8] = damagePanels
 
 
 #########################################
@@ -365,6 +381,7 @@ gspec[2:5, 4:8] = multiDamagePlot
 import matplotlib.pyplot as plt
 
 detImgs = []
+detHists = []
 detGrids = []
 for detImgName in h5["Sums"].keys():
     image = h5[f"Sums/{detImgName}"][()]
@@ -407,14 +424,33 @@ for detImgName in h5["Sums"].keys():
     # between panels!!
     detImgs.append(
         hv.Image(imageR, vdims=[imgDim], name=imgDim.label)
-        .options(colorbar=True, cmap="rainbow")
+        .options(colorbar=True, cmap="viridis")
         .opts(shared_axes=False)
     )
 
+    hist, bins = np.histogram(imageR.flatten(), bins='auto', range=(np.nanpercentile(imageR, 1.0), np.nanpercentile(imageR, 99.0)))
+    histDim = hv.Dimension(
+        ("hist", detImgName.replace("Sums/", "").replace("_calib_img", " Pixel Histogram")),
+        range=(np.nanpercentile(imageR, 1.0), np.nanpercentile(imageR, 99.0)),
+    )
+    detHists.append(
+        hv.Histogram((hist, bins), kdims=[histDim], label=histDim.label).opts(
+            fill_color="#21918c",
+            line_color="black",
+            line_width=0.2,
+            alpha=0.85,
+            title=histDim.label,
+            xlabel="Pixel Intensity",
+            ylabel="Count",
+            invert_axes=True,
+        )
+    )
+
     detGrid = pn.GridSpec(
-        sizing_mode="stretch_both", max_width=700, name=detImgName.replace("Sums/", "")
+        sizing_mode="stretch_both", max_width=1600, max_height=1000, name=detImgName.replace("Sums/", "")
     )
-    detGrid[0, 0] = pn.Row(detImgs[-1])
+    detGrid[0:2, 0:2] = pn.Row(detImgs[-1])
+    detGrid[0:2, 2] = pn.Row(detHists[-1])
     detGrids.append(detGrid)
 
 if nOff > 100:
@@ -464,18 +500,82 @@ if nOff > 100:
         )
         detImgs.append(
             hv.Image(imageR, vdims=[imgOffDim], name=imgOffDim.label).options(
-                colorbar=True, cmap="rainbow"
+                colorbar=True, cmap="viridis"
             )
         )
 
         detGrid = pn.GridSpec(
             sizing_mode="stretch_both",
-            max_width=700,
+            max_width=1600,
+            max_height=1000,
             name="%s, dropped shots" % detName,
         )
         detGrid[0, 0] = pn.Row(detImgs[-1])
         detGrids.append(detGrid)
 
+#################
+# Intensity Plots
+#################
+
+# IPM CORRELATION
+#################
+corrGrids = []
+for detName in detNames:
+    try:
+        azav = h5[f"{detName}/azav_azav"][evtFilter]
+    except:
+        try:
+            azav = h5[f"{detName}/pyfai_azav"][evtFilter]
+        except:
+            continue
+
+    if len(azav.shape) > 2:
+        total_scatt = np.nanmean(azav, axis=(1, 2))
+    else:
+        total_scatt = np.nanmean(azav, axis=1)
+
+    scatterDim = hv.Dimension((f"{detName} Scattering", f"{detName} Scattering"))
+    corrGrid = pn.GridSpec(
+        sizing_mode="stretch_both", max_width=1000, max_height=1000, name=f"{detName} Correlation with IPM"
+    )
+
+    # ip4
+    try:
+        ipmDim = hv.Dimension((f"{detName} DG1", f"{detName} DG1"))
+        ip4_plot = hv.HexTiles(
+            (ipmUpVar, total_scatt), kdims=[ipmDim, scatterDim]
+        ).opts(cmap="viridis", xlabel=ipmDim.label, ylabel=scatterDim.label, title=f"{detName} Scattering vs DG1")
+        corrGrid[0:2, 0:2] = ip4_plot
+    except Exception as e:
+        print(e)
+        ip4_plot = None
+
+    # ip5
+    try:
+        ipmDim = hv.Dimension((f"{detName} DG2", f"{detName} DG2"))
+        ip5_plot = hv.HexTiles(
+            (ipmDownVar, total_scatt), kdims=[ipmDim, scatterDim]
+        ).opts(cmap="viridis", xlabel=ipmDim.label, ylabel=scatterDim.label, title=f"{detName} Scattering vs DG2")
+        corrGrid[0:2, 2:4] = ip5_plot
+    except Exception as e:
+        print(e)
+        ip5_plot = None
+
+
+    try:
+        timeDim = hv.Dimension(("Time", "Time"))
+        time_plot = hv.HexTiles(
+            (eventTimeR, total_scatt), kdims=[timeDim, scatterDim]
+        ).opts(cmap="viridis", xlabel=timeDim.label, ylabel=scatterDim.label, title=f"{detName} Scattering vs Time")
+        corrGrid[2:4, 0:4] = time_plot
+    except Exception as e:
+        print(e)
+        time_plot = None
+
+    if ip4_plot or ip5_plot or time_plot:
+        corrGrids.append(corrGrid)
+
+
 #############
 # XES Plots
 #############
@@ -580,7 +680,7 @@ for det in ["epix100_0", "epix100_1"]:  # detNames:
         )
         # dimg_flip = np.fliplr(dimg)
 
-        xes_grid = pn.GridSpec(max_width=700, name=f"XES (Droplet) - {det}")
+        xes_grid = pn.GridSpec(max_width=1000, max_height=1000, name=f"XES (Droplet) - {det}")
         img_dim = hv.Dimension(("Image", "Image"))
         spatial_dim = hv.Dimension(("Spatial Projection", "Spatial Projection"))
         xes_dim = hv.Dimension(("XES Spectrum", "XES Spectrum"))
@@ -609,7 +709,7 @@ for det in ["epix100_0", "epix100_1"]:  # detNames:
         )
 
         # DO ROI PROCESSING -----
-        xes_roi_grid = pn.GridSpec(max_width=700, name=f"XES (ROIs) - {det}")
+        xes_roi_grid = pn.GridSpec(max_width=1000, max_height=1000, name=f"XES (ROIs) - {det}")
         roi_nums, spatial_projs, xes_projs = processROIData(h5=h5, det_name=det)
         for idx in range(len(roi_nums)):
             roi_num: str = roi_nums[idx]
@@ -647,7 +747,7 @@ for det in ["epix100_0", "epix100_1"]:  # detNames:
 ###########
 # FEE Plots
 ###########
-feeGrid = pn.GridSpec(max_width=700, name="FEE Stats")
+feeGrid = pn.GridSpec(max_width=1000, max_height=1000, name="FEE Stats")
 try:
     feeDamageDim = hv.Dimension(("damage/feespec", "FEE Present"))
     feeDamageVar = h5[feeDamageDim.name][()]
@@ -690,6 +790,9 @@ for xes_grid in xesPlots:
 for detGrid in detGrids:
     tabs.append(detGrid)
 
+for corrGrid in corrGrids:
+    tabs.append(corrGrid)
+
 tabs.append(feeGrid)
 
 if int(os.environ.get("RUN_NUM", "-1")) > 0:
@@ -701,7 +804,7 @@ if int(os.environ.get("RUN_NUM", "-1")) > 0:
 if save_elog:
     from summaries.summary_utils import prepareHtmlReport
 
-    pageTitleFormat = "BeamlineSummary/BeamlineSummary_Run{run:04d}"
+    pageTitleFormat = "BeamlineSummary/{run:04d}"
     prepareHtmlReport(tabs, expname, run, pageTitleFormat)
     postDetectorDamageMsg(
         h5=h5,
